### PR TITLE
Fix generation of front50 config from appengine account

### PR DESCRIPTION
**Pull request by lwander**

### CODE REVIEW

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/Front50Profile.java [hunk 1]
```diff
@@ -19,6 +19,7 @@
 import com.netflix.spinnaker.halyard.config.model.v1.node.Account;
 import com.netflix.spinnaker.halyard.config.model.v1.node.DeploymentConfiguration;
 import com.netflix.spinnaker.halyard.config.model.v1.node.PersistentStorage;
+import com.netflix.spinnaker.halyard.config.model.v1.providers.google.CommonGoogleAccount;
 import com.netflix.spinnaker.halyard.config.problem.v1.ConfigProblemBuilder;
```
Comment (duftler) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/Front50Profile.java]: https://www.youtube.com/watch?v=ueZ6tvqhk8U

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/Front50Profile.java [hunk 2]
```diff
@@ -43,17 +44,20 @@ public SpinnakerArtifact getArtifact() {
...
+
+    if (account instanceof CommonGoogleAccount) {
+      credentials.getSpinnaker().setGcs(new Front50Credentials.Spinnaker.GCS(storage, (CommonGoogleAccount) account));
       config.setRequiredFiles(dependentFiles(account));
       config.extendConfig(config.getPrimaryConfigFile(), yamlToString(credentials));
       return config;
     } else {
-      throw new HalException(
-          new ConfigProblemBuilder(Severity.FATAL, "No valid account configured for persistent storage.").build());
+      throw new RuntimeException("Validation failure: google or appengine account name expected in PersistentStorage configuration.");
```
Comment (duftler) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/Front50Profile.java]: The err msg in `PersistentStorageService` reads "You must supply either a GCE, or Appengine account to your persistent storage service". Are we referring to a Google account or a GCE account?

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/Front50Profile.java (hunk 1 above)
Comment (lwander) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/Front50Profile.java]: Which one is different? I haven't guessed it yet

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/Front50Profile.java (hunk 2 above)
Comment (lwander) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/Front50Profile.java]: Good point, It should read "gce"

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/Front50Profile.java (hunk 1 above)
Comment (duftler) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/Front50Profile.java]: Sort-order is off. This line needs to move down one place.

**Pull request merged**